beacon_node/beacon_chain/src/blob_verification.rs

```diff
@@ -185,8 +176,32 @@ pub fn validate_blob_sidecar_for_gossip<T: BeaconChainTypes>(
         });
     }
 
-    // TODO(pawan): should we verify locally that the parent root is correct
-    // or just use whatever the proposer gives us?
+    // We have already verified that the blob is past finalization, so we can
+    // just check fork choice for the block's parent.
+    if let Some(parent_block) = chain
+        .canonical_head
+        .fork_choice_read_lock()
+        .get_block(&block_parent_root)
+    {
+        if parent_block.slot >= blob_slot {
+            return Err(BlobError::BlobIsNotLaterThanParent {
+                blob_slot,
+                parent_slot: parent_block.slot,
+            });
+        }
+    } else {
+        return Err(BlobError::BlobParentUnknown {
+            blob_root: block_root,
+            blob_parent_root: block_parent_root,
+        });
+    }
+
+    // Note: The spec checks the signature directly against `blob_sidecar.message.proposer_index`
+    // before checking that the provided proposer index is valid w.r.t the current shuffling.
+    //
+    // However, we check that the proposer_index matches against the shuffling first to avoid
+    // signature verification against an invalid proposer_index.
+    // TODO: check if getting the shuffling more expensive than signature verification in any scenario?
     let proposer_shuffling_root = signed_blob_sidecar.message.block_parent_root;
 
     let (proposer_index, fork) = match chain
```

Comment:
> state.get_beacon_proposer_index(blob_slot, &chain.spec)?

If we get the blob before the block, the head state might be at slot - 1, so at an epoch boundary we might hit this: https://github.com/sigp/lighthouse/blob/stable/consensus/types/src/beacon_state.rs#L767

Reply:
Nice catch. Did something more elaborate now to get the proposer index by referring to block verification.
Need to make sure i'm not doing anything stupid playing around with beacon snapshots.